src/Catty/DataModel/Bricks/Data/WebRequestBrick.swift

```diff
@@ -0,0 +1,88 @@
+/**
+ *  Copyright (C) 2010-2020 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+@objc(WebRequestBrick)
+@objcMembers class WebRequestBrick: Brick, BrickProtocol, BrickFormulaProtocol, BrickVariableProtocol {
+
+    var downloader: WebRequestDownloader?
+    var request: Formula?
+    var userVariable: UserVariable?
+
+    override required init() {
+        super.init()
+    }
+
+    func createDownloader(url: String, session: URLSession?) {
+        self.downloader = WebRequestDownloader(url: url, session: session)
```

Comment:
Why are you storing the `downloader` in a property of the Brick? When executed multiple times in parallel this might lead to a problem, or at least is misleading. Suggestion: create and hold it in the instruction it self (does not have to be an instance variable, right?).